### ISSUE TITLE
Switch user

### ISSUE THREAD

**Issue by nucshuco** [plexinc/plex-media-player#11]
When I switch users I need to go through the whole website pin thing. It never sticks like the other apps.

**Comment by tru** [comment]
Sorry, I am not sure what's wrong here. You press switch users in the top right menu item and that takes you to the pin screen instead of the user swisher?

**Comment by nucshuco** [comment]
![1](https://cloud.githubusercontent.com/assets/8262089/10639539/b941c3d6-77c5-11e5-96d4-d9858923909e.JPG)
![2](https://cloud.githubusercontent.com/assets/8262089/10639541/be64886c-77c5-11e5-9261-3de790ef6bb0.JPG)
![3](https://cloud.githubusercontent.com/assets/8262089/10639547/c433f89a-77c5-11e5-8d94-c2ca7e4db480.JPG)

Every time I want to switch user I see the sign in.

**Comment by dogonwheels** [comment]
@kayone - is this PMP only, or something that's appeared on other platforms?

**Comment by nucshuco** [comment]
Only PMP. Work fine with web app, PHT, Plex for window 10.

**Comment by kayone** [comment]
@nucshuco I can't repro this, which platform are you running (windows, osx, version?) also could you give me the version of PMP that you are having this issue with?

can you change any of the settings in the settings menu, restart PMP and see if the changes are still there,

**Comment by nucshuco** [comment]
I'm on win 10
![4](https://cloud.githubusercontent.com/assets/8262089/10645686/e14f8598-77e3-11e5-8137-11d514096b19.JPG)
i can see and edit the setting. When I restart PMP it goes to the last account I selected. If I try to switch users I go to this screen 
![3](https://cloud.githubusercontent.com/assets/8262089/10645782/5f7ece10-77e4-11e5-8c4a-d1caadffc065.JPG)

**Comment by nucshuco** [comment]
Going to install on different pc and see if I can repro this.

**Comment by mikedm139** [comment]
@nucshuco are both/all relevant users members of the same Plex *Home*? Or, are they separate users without a shared *Home*?

**Comment by nucshuco** [comment]
this is from plex web app. hope it helps
![11](https://cloud.githubusercontent.com/assets/8262089/10866294/32f5072a-7fe5-11e5-9550-46eb55b2f9a4.JPG)

**Comment by kayone** [comment]
Ok, I was finally able to reproduce this, what seems to be happening is at times during switch user when we try to get the accessToken out of storage the request fails with 

>[ WARN  ] SettingsSection.cpp @ 69 - Looking for value: "myPlexAccessToken" in section: "webclient" but it can't be found 

looking at `storage.json` I can see the value so there is something that's blocking the app from reading it correctly.

**Comment by dogonwheels** [comment]
@tru - our PMP storage API is async, but are we 100% sure that the write is complete by the time the Qt callback is made to web?

**Comment by tru** [comment]
Hmm. Remind me tomorrow and I'll have a look.

**Comment by wm4** [comment]
Which callback to you mean?

**Comment by kayone** [comment]
we call `channel.api.settings.setValues()` which I assume goes to

https://github.com/plexinc/plex-media-player/blob/manual-server-config/src/settings/SettingsComponent.cpp#L222

**Comment by tru** [comment]
Yeah a quick look tells me that the value should be there when you get the callback on the function call.

**Comment by kayone** [comment]
thanks for the check, I'll do more testing to make sure we aren't racing on our side.

**Comment by mseeley** [comment]
Reopening for another pass. Good luck Keivan.

**Comment by hthighway** [comment]
I am still seeing this issue with then embedded version running on an Intel NUC
